modules/core/04-channel/keeper/packet.go

```diff
@@ -342,14 +342,15 @@ func (k Keeper) WriteAcknowledgement(
 		return types.ErrAcknowledgementExists
 	}
 
-	if len(acknowledgement) == 0 {
+	bz := acknowledgement.Acknowledgement()
```

Comment:
You should still check if acknowledgement is nil first or it will panic

Reply:
Good call, thank you